### ISSUE TITLE
Unable to mount protected Mac paths after upgrade to Docker 3.0.0 

### ISSUE THREAD

**Issue by andyvanosdale** [docker/for-mac#5115]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 822E1E1F-1383-4437-8A4D-4922B6AD72B0/20201210165314

#### Keep v3.0.0
Disable "Use gRPC Fuse for file sharing" under "Experimental Features" in the Docker Preferences.
![image](https://user-images.githubusercontent.com/2337208/101811286-c3591000-3ace-11eb-90ea-c1660bc27439.png)

#### Downgrade
Downgrade to [v2.5.0.1](https://desktop.docker.com/mac/stable/49550/Docker.dmg)

### Expected behavior
I should be able to mount protected Mac paths from `~`

### Actual behavior
Mounts denied
```
$ docker run --rm -it -v ${PWD}:/app alpine
docker: Error response from daemon: Mounts denied: approving /Users/me/Downloads: file does not exist.
ERRO[0000] error waiting for container: context canceled
```

### Information
This is a new problem with 3.0.0. This was working before upgrading to Docker 3.0.0. This happens with any protected path:
* `~/Downloads`
* `~/Desktop`
* `~/Documents`

I am able to mount `~` itself.

`/Users` is listed in File Sharing:
![image](https://user-images.githubusercontent.com/2337208/101804434-6eb19700-3ac6-11eb-9f61-9fd997840a8e.png)

Docker is allowed access to Downloads and Desktop through Security and Privacy:
![image](https://user-images.githubusercontent.com/2337208/101805544-bdabfc00-3ac7-11eb-9baf-886b29558bf7.png)

  - macOS Version: 10.15.7 (19H15)

### Diagnostic logs
```
Docker for Mac: 3.0.0
```

### Steps to reproduce the behavior
1. `cd ~/Downloads`
2. `docker run --rm -it -v ${PWD}:/app alpine`

**Comment by Vardiak** [comment]
This doesn't seem to apply only to Downloads... I cannot mount any project 😓

**Comment by andyvanosdale** [comment]
@Vardiak What paths are you trying to mount? I can't mount any protected Mac paths under /Users, like Documents, Downloads, or Desktop.

**Comment by Vardiak** [comment]
@andyvanosdale Any project under /Users/Documents. What is a protected Mac path ? Is there a workaround ?

**Comment by andyvanosdale** [comment]
I'm trying to find a workaround...

**Comment by Ben-Atherton** [comment]
I've got the same issue, I can't mount anything under /Users/Ben/Documents

I've managed to get it working again (for now) by downgrading to v2.5.0.0 which can be downloaded at https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2500

**Comment by andyvanosdale** [comment]
Relevant [Apple Support article](https://support.apple.com/guide/mac-help/control-access-files-folders-mac-mchld5a35146/10.15/mac/10.15) about folder privacy

**Comment by jfreed** [comment]
Having the same issue. Disabling "Use gRPC FUSE for file sharing" fixes it.

**Comment by ericmacfa** [comment]
I'm running into this issue as well, and I think it's worth noting that I have indeed set up my Mac's permissions to allow Docker access to my Documents folder.
![image](https://user-images.githubusercontent.com/18381801/101810608-9ca2d580-3ade-11eb-9ee7-b465d896ac77.png)

**Comment by andyvanosdale** [comment]
If gRPC Fuse is the culprit, I suspect its some other process doing the mounting that Mac isn't able to expose in the UI.

**Comment by andyvanosdale** [comment]
Hat tip to @jfreed. Can confirm disabling gRPC Fuse is a workaround. I've updated the description with the details.

**Comment by stephen-turner** [comment]
Thanks for the report. I think this is a side effect of fixing #4975 where people were scared when we asked to see Reminders and Downloads. It looks like we need to think again about the solution.

**Comment by alfredotranchedone** [comment]
I confirm that disabling "Use gRPC FUSE for file sharing" fixes the problem

**Comment by adamoki** [comment]
Is there any impact on performance when disabling "Use gRPC FUSE for file sharing"? Anybody can give input on that?

**Comment by sfcgeorge** [comment]
Tried uninstalling and reinstalling 3.0 but that didn't help. Tried manually adding permissions in Security & Privacy but that didn't help either. 93EE1AF6-928B-4DD0-9DA9-6ABAF23380D2/20201210181155

As I believe FUSE is the default and I imagine most people use bind mounts, it might be worth pulling the 3.0 update until this is fixed.

**Comment by Vardiak** [comment]
> I confirm that disabling "Use gRPC FUSE for file sharing" fixes the problem

When disabling gRPC FUSE, I can't run a container for the second time with the same path shared, I have to restart the daemon every time...

**Comment by delfuego** [comment]
> As I believe FUSE is the default and I imagine most people use bind mounts, it might be worth pulling the 3.0 update until this is fixed.

I kind of agree with this — this is a big breakage, and I've had to manually walk a few of our developers through the downgrade back to 2.5.0.0 in the few hours since this hit.

**Comment by glebovdev** [comment]
Downgrade to v. 2.5.0.1 (https://desktop.docker.com/mac/stable/49550/Docker.dmg) also possible, no issues so far.

**Comment by aniket-patel-ef** [comment]
I am also facing the same issue.. Thanks to this thread there is a workaround after hours of debugging

**Comment by folt** [comment]
I faced the same problem. I cannot view any volumes in my project.

**Comment by alphonse92** [comment]
Same,  Disabling "Use gRPC FUSE for file sharing" fixes it.

**Comment by rstropoli** [comment]
Same here, Disabling "Use gRPC FUSE" fixed the issue.

**Comment by pleminh** [comment]
Same problem . 
Disable gRPC FUSE fixes it . 

` ERROR: for project-app  Cannot start service project-app: Mounts denied: approving`

**Comment by chaznied** [comment]
disabling gRPC FUSE worked for me

**Comment by guice** [comment]
Ditto. Had to disable gRPC. I couldn't mount a folder within `/Users/{me}/Documents` even though `/Users` is shared, and Docker has rights to `Documents/`

**Comment by ricardorodriguezg** [comment]
The first workaround worked for me, thanks!
Disable "Use gRPC Fuse for file sharing"

**Comment by JoaoSobral** [comment]
I have downgraded. But just to report the same situation. I  kept looking at my path for 10 minutes ()-()-  "software error messages are most always right they say..."

**Comment by surajrav** [comment]
+1 here. I'm glad to have found this issue on github because I was starting to doubt my own sanity. Thanks @andyvanosdale and all those who reported on this thread! Add my mac console log screenshot if it helps:
![image](https://user-images.githubusercontent.com/8673958/101848950-8c551f80-3b0b-11eb-9948-8ab3d195fe22.png)

**Comment by jooyoungho** [comment]
Disable "Experimental Features -> Use gRPC FUSE for file sharing"
Thank you!

**Comment by linhnvhicts** [comment]
I was wondering why are of my docker container stopped working, Please don't enable this feature by defaults because it's breaking on MacOS after updating

**Comment by jgoodie** [comment]
Confirmed disabling gRPC helps, but restarting a container with the same path requires a restart of the daemon.

**Comment by haozzzzzzzz** [comment]
Me too. I cannot mount any file path under /Users. I am crying......

**Comment by 39ff** [comment]
same problem.
I can't mount /Users

**Comment by ylbeethoven** [comment]
Time to downgrade, this new update is such a mess. ALL of my docker compose files stop working after this update. 😥

**Comment by komorin0521** [comment]
I got the same problem but it's work.Thanks!!

**Comment by EthanYan6** [comment]
> Having the same issue. Disabling "Use gRPC FUSE for file sharing" fixes it.

Thanks, useful, solved

**Comment by llitfkitfk** [comment]
![image](https://user-images.githubusercontent.com/2404785/101880064-482a5380-3bcd-11eb-8e73-b72fc7fee2c7.png)
diable "Use gRPC FUSE for file sharing"

**Comment by tristansb1** [comment]
For people that still have issues after disabling `Use gRPC FUSE for file sharing` try restarting your mac...

**Comment by osenv** [comment]
This is really annoying! Very disappointed with this upgrade.

**Comment by bastos71** [comment]
With the "disable use gRPC FUSE" workaround I've got big issues with files syncing (ie : files build via `npm run watch`).
Downgrading to previous version seems the best solution

**Comment by 42robotsCS** [comment]
Same here. Cannot mount files under  /Users/<username>/Documents/<projects_directory>.
Also every path is set in Docker Desktop.
Experienced the issue after upgrading to Docker Desktop for Mac 3.0.0.
Rolling back to 2.5 for now.

**Comment by Mon-Tea** [comment]
Same, but Disabling "Use gRPC FUSE for file sharing fixed it 
Also ddev works again …

**Comment by zeornelas** [comment]
I've also experienced this problem. I had to roll back to 2.5.0.

**Comment by osenv** [comment]
Just rolled back to 2.5.0 as well.

**Comment by bilhackmac** [comment]
One of my colleague have exact same problem as all of you… but me not !

+ We are both under BigSur
+ File sharing under Docker For Mac settings are same
+ Terminal (iTerm for me) with full disk access (but not docker) in system security settings
+ Docker with all needed local folder access (`~/Downloads`, `~/Desktop`, `~/Documents`, `Removable volumes`) in system security settings
+ _Use gRPC FUSE for file sharing_ IS active

I don't know why it works for me and not for you.

If you want more details, ask me, I don't have much time but I'll try to answer ASAP

**Comment by furazin** [comment]
Hi! You saved my morning today, disabling the **gRPC FUSE for file sharing** option and restarting works again without downgrading.
Thanks.

**Comment by dlgoodchild** [comment]
I've just hit this problem on Catalina and thankfully the top google result for "docker Mounts denied: approving" is this thread! Silly to release such a breaking change that probably affects many people's work.
Anyway, disabling "**Use gRPC FUSE for file sharing**" has fixed it for me for the moment. When I've more time I'll look into seeing what's needed to keep it enabled.

**Comment by stephen-turner** [comment]
It took a little longer than I predicted, but this is now fixed. If you do a Check for Updates from the whale menu, you can get the new version (3.0.1) as a delta update.

**Comment by jfreed** [comment]
Can confirm the update fixed the issue for me.

**Comment by stephen-turner** [comment]
Thanks @jfreed, that's good to know.

By the way, ~/Library is still blocked. We don't see a use case for sharing that, and it will cause problems to do so because the docker logfiles are written there.

**Comment by alexmanzo** [comment]
+1 Confirming update works with "Use gRPC FUSE for file sharing" re-enabled.

**Comment by andyvanosdale** [comment]
Also confirmed it is resolved! I have all the protected folders listed in the Files and Folder Privacy section, so I'm not able to confirm that when I try to access a folder not in that list I will get prompted by macOS. If anyone is able to verify this, that would be great!

**Comment by roy-work** [comment]
> By the way, ~/Library is still blocked. We don't see a use case for sharing that

My team writes a tool, that, to ensure that its runs are reproducible to the greatest extent possible, re-executes itself inside a Docker container. However, it caches some data (a large git repository) between executions on the host user's FS in `~/Library/Caches/<our org>/`.

**Comment by s3f4** [comment]
> +1 Confirming update works with "Use gRPC FUSE for file sharing" re-enabled.

**Comment by skrzypek** [comment]
I do have all my repositories at iCloud under `/Users/<user>/Library/Mobile Documents/com~apple~CloudDocs` path, I still hit this bug.

**Comment by kiziltas** [comment]
Had the same issue, tried the workaround shared here and restarted docker but this time it didn't start properly (hanging in starting), updated to 3.0.1 and still NOT booting.

**Comment by alextsayun** [comment]
3.0.1 is working fine for me, but only after "reset to factory defaults".

**Comment by ConsiderCode** [comment]
Can confirm that 3.0.1 fixed this and I did not need to reset to factory defaults. Thank you @andyvanosdale for raising this issue so quickly and suggesting a work around. It saved us a lot of downtime.

**Comment by aditilamkhede** [comment]
Yes, update fixed the issue for me too(without disabling "Use gRPC FUSE for file sharing").

**Comment by jmaroeder** [comment]
Count me as one more vote for the ability to share files from inside `~/Library` (or at least document in the release notes that this behavior is unsupported).

My argument for this is that `~/Library/Application Support` is the default "application data" directory. If I want to mount data that is generated by other applications running on my system, I either have to copy it out of the default location or deal with some symlinks (and I'm not sure if those are supported).

**Comment by stephen-turner** [comment]
Thanks for the use cases for (subdirectories of) `~/Library`. Obviously this was an urgent fix, and we decided that this was the best compromise that we could do quickly. We do intend to return to it and try and find a way to fix this ticket and #4975 simultaneously.

**Comment by dee-kryvenko** [comment]
> By the way, ~/Library is still blocked. We don't see a use case for sharing that, and it will cause performance problems to do so because the docker logfiles are written there.

> Thanks for the use cases for (subdirectories of) `~/Library`

I don't want to appear ungrateful for your work or something, but I'm very troubled and concerned while trying to wrap my head around the very idea that somebody even had to give you a use case for something like that. I mean, what makes you think that Docker should be in position to be making decisions as to what user can and can't mount to his own containers on his own system? What was the use case to even have this filtering implemented with no ability for the user to have items added and removed from it in the first place? This seems like a huge overreach from the Docker side and it troubles me very much - like what's next going to be limited for the sake of user's own protection, maybe running unauthorized containers from other than Docker Hub sources should be disallowed?

**Comment by dee-kryvenko** [comment]
To add some weight, I want to point out that the entire DevOps community (or at least these of us who is trying to do CICD in a containerized way, which is pretty much most of us this days), are very affected right now even with `3.0.1` fix. This is so common for portability purposes to move the entire toolchain into the containers and execute them as containers on our local workstations. It's a great deal of portability and it's crucial. And it's so happens that directories that the user would want to mount to his containers in this scenario are all over the place, `~/.aws` and `~/.kube` to begin with. Most of them happen to be under `~/Library`, build caches are a huge part of it - Go included which happens to be in `~/Library/Caches/go-build`.

**Comment by dee-kryvenko** [comment]
And to add some context (and pardon me for the multiple messages, but they are different aspects of a complex issue and to me they are appear to be logically separate), I would like to point out that there are two major use cases for Docker.

**Isolation** is one of them and it seems to have the most attention. Surely enough the ability to run applications securely in an isolated way is crucial and it had dramatic impact and completely transformed Software Engineering as we know it today.

But **Portability** is another one that is often remains in the shadow of the first one, yet should not be underestimated as both of them are yin and yang of the containerized world. Ability to distribute and run not just applications but entire ecosystems at our laptops that guarantees the execution environment across the different users and automated systems is a big deal. One could envision the world with no apt, yum, brew, chocolatey and such where all the software basically distributed as containers - and it would be hard to get to that world if portability principal is neglected in favor of total isolation.

It absolutely and ultimately has to be a user choice as to how much security does he need, since there are absolutely different use cases. Like if I run just the app I'm developing - it's one thing, but what if I run something like Terraform or Helm in my containers and I do intentionally want it to have access to parts of my system that in other scenarios might be considered a vulnerability?

**Comment by dlgoodchild** [comment]
@dee-kryvenko I think you could be jumping the gun here on all this.

It _is_ an experimental feature afterall. Disable it if you need `~/Library`.

There's clearly a technical reason behind not being able to use `~/Library` with gRPC FUSE, so maybe they'll simply address that when there's an evident demand for it and maybe when it's not experimental. Remember, they have to prioritise work, and if this isn't a priority then they can leave it "blocked" and address other more pressing issues.
They could just not expose experimental features until they are fully supported. They are afterall contending with different technologies, and a more restrictive OS than most others. Just cos something is blocked for the time being, it doesn't necessarily imply it's some kind of draconian overreach by Docker-for-mac.

**Comment by tejksat** [comment]
@dlgoodchild please correct me if I am wrong but despite being experimental "Use gRPC FUSE" feature [is enabled by default](https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2400) for Docker for Mac:
> Docker Desktop now uses gRPC-FUSE for file sharing by default.

@stephen-turner for us in JetBrains this issue keeps affecting our users in 3.0.1 on attempts to run tests with coverage within the Docker integration in PyCharm, RubyMine and PhpStorm  (f.e. please see [this issue](https://youtrack.jetbrains.com/issue/RUBY-27508)) as we bind `~/Library/Caches/JetBrains` subdirectories to the container to retrieve the coverage data.

Unfortunately, as "Use gRPC FUSE" feature is enabled by default, our users experience failures in previously working scenarios. This evidently frustrates users and we only could advise them to disable this feature in Docker for Mac as soon as they reach us (until we prepare the fix and what is even more crucial, until they update their IDEs).

Is it possible to include either `~/Library` or a subset of it including `~/Library/Caches` to the list of shared paths? Or f.e. disable this feature by default for a transitional period of time with the warning emerging when someone tries to bind mount directories that will not be available when this feature becomes enabled?

**Comment by stephen-turner** [comment]
The reason we disabled it in the first place was because of scary security warnings as reported in #4975. We are trying to work round that, but as they are OS level messages we don't have much control over them. There are also performance concerns if Docker can sync its own working directories because of the risk of infinite loops, but those are more under our control.

We realise the current situation is not satisfactory for people who do want to mount subdirectories of `~/Library`, even if it has fixed #4975 for the much larger number of people who blindly mount the whole of `~`. So today we are looking at (1) can we exclude an even smaller set of directories (but I bet there's someone somewhere who wants to mount `~/Library/Reminders` for some reason); (2) can we only mount them if they are explicitly read from within the container (and can we do that in a performant way?).

**Comment by dlgoodchild** [comment]
> @dlgoodchild please correct me if I am wrong but despite being experimental "Use gRPC FUSE" feature [is enabled by default](https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2400) for Docker for Mac:

You're absolutely right, and that is probably the root of the problem. Who in their right mind would make an experimental feature the default? It's crazy that it just slips in there and breaks everything. Maybe it should have been a prompt post-install like "We recomment this feature, but it's experimental ....blah blah".

**Comment by dee-kryvenko** [comment]
@dlgoodchild I do not think I was overreacting. First of all as already being mentioned - labeling this feature experimental means nothing to me as experimental features enabled by default does not sound so experimental to me.

But what's even more important and what's not being mentioned so far - we all know how it goes with all the experimental features. They eventually becomes a new norm and the old norm being slowly pushed out and retired. Even if they might not have the intention to retire the old mechanism right now, but as you mentioned yourself - it always comes to prioritizing limited resources so I do not think the old mechanism will be getting too much attention going forward and will naturally be retired in a some time.

And the fun part - I did tried to disable gRPC FUSE on 3.0.1 and I faced another issue - files updated from the host was not updating in the container. Had to downgrade since I don't have time to investigate this issue and file a new ticket right now, it might very well be that I just faced another known issue like https://github.com/moby/moby/issues/15793.

Whether there are technical reason behind not being able to use `~/Library` with gRPC FUSE or not seems irrelevant to me. Docker can't not be having an ability to mount `~/Library`, it would just make docker completely unusable. This should be either fixed or gRPC FUSE is just not the right solution here.

And finally, adding weight and trying to indicate exactly how much there is an evident demand for this out there and how extreme the priority is - is exactly what I am trying to do here. It might be a little too late to be doing that when this not-so-experimental feature becomes not-at-all-experimental.

@stephen-turner, quite honestly I am not even sure why https://github.com/docker/for-mac/issues/4975 was even considered. This is clearly an OS issue. You mount stuff - OS asking permissions. It's just the way it is. Maybe it's just me but I would expect Docker users to know exactly why such a warnings pops up. It's not like Docker users are sales managers and bank clerks, or are they? We all here getting the basic concepts of OS and FS and security permissions...

Anyway, whatever you decide to do with it - just remember you can't be making assumptions like "no, the users certainly would never need to mount this particular directory". They most certainly will. Current behavior is not just "not satisfactory" - it's making Docker completely useless for a subset of DevOps community that packs their toolchain as docker images - and how big exactly that subset is I don't know but I want to believe it's an ultimate majority at the very least.

**Comment by stephen-turner** [comment]
It is absolutely our intention to replace osxfs with gRPC Fuse, and we have stated so many times. gRPC Fuse is now the default because we consider it superior overall, although they each have issues that the other doesn't have, which is why we have retained the toggle for the moment. To be honest it probably shouldn't be in the experimental panel any more now that it's graduated to be the default, but we don't have a better place for it and so kept it there.

The symptom with #4975 is that someone bind mounts the whole of `$HOME` — and not even typing it explicitly, they are using a web framework that does that internally — maybe they don't even understand Docker very well but just know the runes to make their project compile — and then _several hours later, when they're not even using Docker_, a warning pops up saying that Docker wants to read their Reminders. This is scary, and they don't understand it. So it absolutely needs to be fixed.

Anyway, I think I've been clear that we're fixing this as fast as we can. We just need to find the right solution that doesn't make things worse for the much larger web developer audience. Our current plan is what I listed as (3) above: `~/Library` (or a subdirectory) would be excluded by default if you only share the home directory, but you can explicitly include it to be shared.

**Comment by dee-kryvenko** [comment]
To be honest I probably do not understand the proposed solution, because if I do - it sound to me like Docker is going to have in that case an include list, from which there's going to be exclude list, which in turn will have another include sub-list. That doesn't feel right for a mature product... not very intuitive. Most likely error prone and will cause you troubles in the future.

Why wouldn't docker just pro-actively explain it's behavior to the user? Docker already has a warning with explanation why the user gonna need to enter credentials in the next screen. Docker can just expand on this existing pattern and proactively detect that some of the user actions may trigger these OS warnings and then pop up a warning with explanation that that could happen and the reasons why. This would sound the least intrusive approach.

**Comment by DanielSchetritt** [comment]
Updating to 3.0.1 didn't fix this for me.

**Comment by norrisjeremy** [comment]
I too have to downgrade from 3.0.1 to 2.5.0.1 in order to have functioning bind mounts again.

**Comment by vincentmrg** [comment]
If I understand correctly we have the choice between the version which jokes on the volume mounts and the version with the hyperkit which eats all the cpu.

**Comment by stannnous** [comment]
Confirming this is still broken in 3.0.1. The disable "Use gRPC Fuse for file sharing" workaround continues to work though.

**Comment by n3dj0** [comment]
> Confirming this is still broken in 3.0.1. The disable "Use gRPC Fuse for file sharing" workaround continues to work though.

+1, it is still broken and a factory reset did also NOT help.

**Comment by stephen-turner** [comment]
We don't need any more confirmations if it's `~/Library` that is blocked. (If anything else is still blocked, then please do let us know, but we don't think that's the case).

Our plan has changed since my last message. We're now planning to do what @dee-kryvenko suggested at https://github.com/docker/for-mac/issues/5115#issuecomment-744598956 (thank you, Dee), and issue a warning if someone tries to share the whole of their home directory. I expect it will annoy quite a lot of people as this is a common idiom, especially if they're not sharing it directly but using a framework that shares it, but on reflection it seems like user education probably is the way forward.

**Comment by norrisjeremy** [comment]
> We don't need any more confirmations if it's `~/Library` that is blocked. (If anything else is still blocked, then please do let us know, but we don't think that's the case).
> 
> Our plan has changed since my last message. We're now planning to do what @dee-kryvenko suggested at [#5115 (comment)](https://github.com/docker/for-mac/issues/5115#issuecomment-744598956) (thank you, Dee), and issue a warning if someone tries to share the whole of their home directory. I expect it will annoy quite a lot of people as this is a common idiom, especially if they're not sharing it directly but using a framework that shares it, but on reflection it seems like user education probably is the way forward.

The issue I am having is that I am no longer able to reliably use `qemu-nbd` + `kpartx` to mount disk image files inside a container that sit in bind volumes & this occurs regardless of the gRPC Fuse setting in 3.x.

I frequently end up with invocations of `qemu-nbd` to attach disk image files from within a bind volume that hang forever, or if the `qemu-nbd` invocation randomly succeeds, an invocation of `kpartx` to find the partitions in the disk image fail with `read error, sector 0` + `llseek error`.

This does not occur with 2.5.0.1 (both with gRPC Fuse enabled or disabled).

**Comment by stephen-turner** [comment]
OK, this should be completely fixed in 3.0.2. You can now share ~/Library again.

**Comment by souravchandra** [comment]
Docker desktop is stuck during start after I upgrade to 3.0.2
![image](https://user-images.githubusercontent.com/1384463/102682511-111cf900-41f0-11eb-83f7-977437d68f72.png)

Mac version
![image](https://user-images.githubusercontent.com/1384463/102682520-209c4200-41f0-11eb-9075-5b084dfd30c4.png)

**Comment by dee-kryvenko** [comment]
@stephen-turner seems 3.0.2 is working for me again. Didn't see any warnings though so far.
